src/PlaywrightSharp/FrameManager.cs

```diff
@@ -81,7 +83,7 @@ internal void FrameCommittedNewDocumentNavigation(string frameId, string url, st
 
             if (!initial)
             {
-                foreach (var watcher in LifecycleWatchers)
+                foreach (var watcher in LifecycleWatchers.ToArray())

```

Comment:
Why do we need `ToArray`?

Reply:
in case some other event modifies `LifecycleWatchers` while we're iterating over it. this way it won't throw an error